gpio/gpio.h

```diff
@@ -0,0 +1,98 @@
+//
+// Created by aaron on 8/19/22.
+//
+
+#ifndef LAUNCH_CORE_GPIO_H
+#define LAUNCH_CORE_GPIO_H
+
+
+class gpio {
```

Comment:
for GPIO I was thinking it should be a device type since it's platform specific how this is actually implemented. I'm not sure if we have one device per pin or a general "GPIO controller" or something else entirely but that's where this code should belong. We basically want a device interface (look at device/StreamDevice.h for an example) so that we can implement a GPIO device for whatever platform we want